maestro-orchestra-models/src/main/java/maestro/orchestra/Commands.kt

```diff
@@ -423,6 +423,25 @@ data class AssertWithAICommand(
     }
 }
 
+data class AssertVisualCommand(
+    val baseline: String,
+    val thresholdPercentage: Int,
+    override val optional: Boolean = false,
+    override val label: String? = null,
+) : Command {
+    override fun description(): String {
+        return label ?: "Assert visual difference with baseline $baseline (threshold: $thresholdPercentage%)"
+    }
+
+    override fun evaluateScripts(jsEngine: JsEngine): Command {
+        return copy(
+            baseline = baseline.evaluateScripts(jsEngine)
+        )
+    }
+}
+
+
```

Comment:
Can you add an integration test for this command for the different cases you expect?